fortanix-vme/eif-tools/src/bin/ftxvme-runner-cargo.rs

```diff
@@ -0,0 +1,263 @@
+    }};
+}
+
+static ARGS: Lazy<Vec<String>> = Lazy::new(|| {
+    std::env::args().collect::<Vec<_>>()
+});
+
+#[derive(Debug, Error)]
+enum CommandFail {
+    #[error("Failed to run {0}: {1}")]
+    Io(String, io::Error),
+    #[error("While running {0} got exit status {1}")]
+    Status(String, ExitStatus),
+}
+
+fn run_command(mut cmd: Command) -> Result<(), CommandFail> {
+    match cmd.status() {
+        Err(e) => Err(CommandFail::Io(format!("{:?}", cmd), e)),
+        Ok(status) if status.success() => Ok(()),
+        Ok(status) => Err(CommandFail::Status(format!("{:?}", cmd), status)),
+    }
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(rename_all = "kebab-case")]
+struct CargoTomlMetadata {
+    fortanix_vme: FortanixVmeConfig,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(rename_all = "kebab-case", default, deny_unknown_fields)]
+#[rustfmt::skip] // contains long lines because of links and they may wrapped by mistake
+/// This config is mainly intended for args of ftxvme-elf2eif and nitro-cli run-enclave.
+/// See their args documentation for more info about these opts:
+/// https://docs.aws.amazon.com/enclaves/latest/user/cmd-nitro-run-enclave.html#cmd-nitro-run-enclave-options
+/// https://docs.aws.amazon.com/enclaves/latest/user/cmd-nitro-build-enclave.html#cmd-nitro-build-enclave-options
+struct FortanixVmeConfig {
+    /// Enables verbose mode of ftxvme-elf2eif.
+    verbose: bool,
+
+    /// Path to output eif file
+    eif_file_path: PathBuf,
+
+    /// Path to resources, default is `/usr/share/nitro_enclaves/blobs/`.
+    /// See blobs in: https://github.com/aws/aws-nitro-enclaves-cli#source-code-components
+    resource_path: Option<PathBuf>,
+
+    /// Path to signing certificate. If this is specified,
+    /// `private-key` needs to be specified too.
+    signing_certificate: Option<PathBuf>,
+
+    /// Path to private key. If this is specified,
+    /// `signing-certificate` needs to be specified too.
+    private_key: Option<PathBuf>,
+
+    /// A custom name given to the enclave. If not specified,
+    /// the name of the .eif file is used.
+    enclave_name: Option<String>,
+
+    /// Specifies the number of vCPUs to allocate to the enclave.
+    /// Shouldn't coexist with `cpu_ids`.
+    cpu_count: Option<isize>,
+
+    /// Specifies the IDs of the vCPUs to allocate to the enclave.
+    /// Shouldn't coexist with `cpu_count`.
+    cpu_ids: Option<String>,
+
+    /// Specifies the amount of memory (in MiB) to allocate to the enclave.
+    /// Should be at least 64 MiB.
+    memory: isize,
+
+    /// Basically this field represents the vsock address of
+    /// the enclave.
+    /// This field should always be greater than equal to 4.
+    ///
+    /// The CID of parent instance is always 3 as per:
+    /// https://docs.aws.amazon.com/enclaves/latest/user/nitro-enclave-concepts.html#term-socket
+    ///
+    /// Vsock CID is analogous to an IP address.
+    enclave_cid: Option<String>,
```

Comment:
Similar as above. We might want to have flexibility in choosing the cid in the future.